lib/Doctrine/ORM/Tools/Pagination/Paginator.php

```diff
@@ -124,11 +100,9 @@ public function count()
     /**
      * {@inheritdoc}
      *
-     * @return ArrayIterator
      * @psalm-return ArrayIterator<array-key, T>
      */
-    #[ReturnTypeWillChange]
-    public function getIterator()
+    public function getIterator(): ArrayIterator
```

Comment:
It was already an ArrayIterator before, but should this method rely on an `Iterator` implementation instead of the interface itself. Even the name of the method doesn't refer to a specific Iterator implementation.

It's not really a blocker for a merge, but food for thoughts for 3.0.

Reply:
No, no, I totally see what you mean: if the consumer don't leverage it, why should we provide that level of detail. Here, they shouldn't be calling `getIterator()` directly but use a `foreach`, so that's one more reason to do this. Let me change that.

Reply:
Yes, `IteratorAggregate` allows to access `getIterator()` directly. Technically it should be `Traversable` as the docs suggests, but I wasn't sure if this step was a too large for this PR.